src/main/java/org/spongepowered/common/data/SpongeDataManager.java

```diff
@@ -347,6 +348,7 @@ public static void finalizeRegistration() {
         } else {
             throw new IllegalStateException("Already registered the DataTranslator for " + objectClass.getCanonicalName());
         }
+        TypeSerializers.getDefaultSerializers().registerType(translator.getToken(), DataTranslatorTypeSerializer.from(translator));
```

Comment:
You will need to check that the translator is not registering an already registered type (such as the one for `UUID` which is default registered by Configurate).